FunctionalTableData/TableSectionHeaderFooter.swift

```diff
@@ -15,9 +15,16 @@ public typealias TableHeaderConfigType = TableHeaderFooterConfigType
 
 public protocol TableHeaderFooterConfigType: TableItemConfigType {
 	func dequeueHeaderFooter(from tableView: UITableView) -> UITableViewHeaderFooterView?
+	func isEqual(_ other: TableHeaderFooterConfigType?) -> Bool
 	var height: CGFloat { get }
 }
 
+public extension TableHeaderFooterConfigType {
+	func isEqual(_ other: TableHeaderFooterConfigType?) -> Bool {
+		return (other as? Self) != nil
```

Comment:
What is happening here, this is not checking or equality

Reply:
I didn't want to just add `isEqual(...)` to the protocol without a default implementation as it would become a breaking change; classes no longer conforming to it. The default is only checking that the headers are of the same type.

Reply:
We should still fix this everywhere we have it (and maybe explain to people the impact of putting dynamic things in the header and footer)